src/mms_client/utils/multipart_transport.py

```diff
@@ -0,0 +1,247 @@
+"""Contains a transport override supporting MTOMS attachments."""
+
+import string
+from base64 import b64encode
+from email.encoders import encode_7or8bit
+from email.mime.application import MIMEApplication
+from email.mime.base import MIMEBase
+from email.mime.multipart import MIMEMultipart
+from logging import getLogger
+from random import SystemRandom
+from typing import Dict
+from typing import Optional
+
+from lxml import etree
+from lxml.etree import _Element as Element
+from pendulum import now
+from requests import Session
+from zeep.cache import VersionedCacheBase
+from zeep.transports import Transport
+from zeep.wsdl.utils import etree_to_string
+
+# Set the default logger for the MMS client
+logger = getLogger(__name__)
+
+
+# Define the namespaces used in the XML
+XOP = "http://www.w3.org/2004/08/xop/include"
+XMIME5 = "http://www.w3.org/2005/05/xmlmime"
+FILETAG = "xop:Include:"
+ID_LEN = 16
+
+
+# Define a function to generate a randomized ID string
+def get_id(length: int = ID_LEN) -> str:
+    """Generate a randomized ID string.
+
+    Arguments:
+    length (int):   The length of the ID string to generate.
+
+    Returns:    The randomized ID string.
+    """
+    return "".join(SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(length))
+
+
+def now_b64():
+    """Return a base64 encoded string of the current timestamp."""
+    return b64encode(f"{now().timestamp()}".replace(".", "").encode("UTF-8")).decode("UTF-8")
+
+
+def get_boundary() -> str:
+    """Return a randomized MIME boundary string."""
+    return f"MIMEBoundary_{now_b64()}".center(33, "=")
+
+
+def get_content_id() -> str:
+    """Return a randomized MIME content ID string."""
+    return f"<{now_b64()}@electroroute.co.jp>"
+
+
+def overwrie_attachnode(node):
+    """Overwrite the attachment node.
+
+    Arguments:
+    node (Element):    The XML node to be overwritten.
+
+    Returns:    The attachment node.
+    """
+    cid = node.text[len(FILETAG) :]
+    node.text = None
+    etree.SubElement(node, f"{{{XOP}}}Include", nsmap={"xop": XOP}, href=f"cid:{cid}")
+    return cid
+
+
+def get_multipart(content_id: str) -> MIMEMultipart:
+    """Create a MIME multipart object.
+
+    Arguments:
+    content_id (str):   The content ID to be used for the attachment.
+
+    Returns:    The MIME multipart object.
+    """
+    part = MIMEMultipart(
+        "related", charset="UTF-8", type="application/xop+xml", boundary=get_boundary(), start=content_id
+    )
+    part.set_param("start-info", "application/soap+xml")
+    return part
+
+
+def get_envelopepart(envelope: Element, content_id: str) -> MIMEApplication:
```

Comment:
get_envelope_part

Reply:
I can see the underscore is a bit weak with this one